riscv/src/register/dcsr.rs

```diff
@@ -0,0 +1,214 @@
+//! dcsr register — Debug Control and Status Register (0x7b0)
+//!
+//! Provides control and status for debug mode, including cause of entry, step control, and privilege level.
+
+read_write_csr! {
+    /// Debug Control and Status Register
+    Dcsr: 0x7b0,
+    mask: 0xffff_ffff,
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// Previous privilege level when entering debug mode (bits 0..2)
+    prv: [0:1],
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// Single step mode (bit 2)
+    step: 2,
+}
+
+read_only_csr_field! {
+    Dcsr,
+    /// Non-maskable interrupt pending (bit 3)
+    nmip: 3,
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// Use mstatus.mprv in debug mode (bit 4)
+    mprven: 4,
+}
+
+read_only_csr_field! {
+    Dcsr,
+    /// Cause for entering debug mode (bits 6..8)
+    cause: [6:8],
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// Stop timer increment in debug mode (bit 9)
+    stoptime: 9,
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// Stop counter increment in debug mode (bit 10)
+    stopcount: 10,
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// Interrupt enable during single-step (bit 11)
+    stepie: 11,
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// EBREAK behavior in User mode (bit 12)
+    ebreaku: 12,
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// EBREAK behavior in Supervisor mode (bit 13)
+    ebreaks: 13,
+}
+
+read_write_csr_field! {
+    Dcsr,
+    /// EBREAK behavior in Machine mode (bit 15)
+    ebreakm: 15,
+}
+
+read_only_csr_field! {
+    Dcsr,
+    /// Debug version (bits 28..31)
+    xdebugver: [28:31],
+}
+
+/// Cause for entering debug mode
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum DcsrCause {
+    None = 0,
+    Ebreak = 1,
+    Trigger = 2,
+    HaltRequest = 3,
+    Step = 4,
+    ResetHaltRequest = 5,
+}
+
+impl DcsrCause {
+    pub fn from_usize(val: usize) -> Result<Self, usize> {
+        match val {
+            0 => Ok(Self::None),
+            1 => Ok(Self::Ebreak),
+            2 => Ok(Self::Trigger),
+            3 => Ok(Self::HaltRequest),
+            4 => Ok(Self::Step),
+            5 => Ok(Self::ResetHaltRequest),
+            other => Err(other),
+        }
+    }
+}
+
+/// Previous privilege level when entering debug mode
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum DcsrPrv {
+    User = 0,
+    Supervisor = 1,
+    Machine = 3,
+}
+
+impl DcsrPrv {
+    pub fn from_usize(val: usize) -> Result<Self, usize> {
+        match val {
+            0 => Ok(Self::User),
+            1 => Ok(Self::Supervisor),
+            3 => Ok(Self::Machine),
+            other => Err(other),
+        }
+    }
+}
```

Comment:
As mentioned before, instead of manually implementing all these enumerations, you can use the `csr_field_enum!` macro